src/plug/bash_plugin.rs

```
@@ -1,40 +1,125 @@
-use std::process::Command;
-use std::path::{Path,PathBuf};
+use crate::plug::{unwrap_home_path, CallablePlugin, PluginInterface, PLUGINS_PATH};
+use chrono::Local;
 use std::collections::HashMap;
-use crate::plug::{unwrap_home_path, PluginInterface, CallablePlugin, PLUGINS_PATH};
-
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use std::process::{Child, ChildStdout, Command, Stdio};
 
 struct BashCommand {
     executable: PathBuf,
 }
 
+enum ProcessStatus {
+    Running,
+    Finished(i32),
+}
+
+
+fn get_process_status(child: &mut Child) -> ProcessStatus {
+    match child.try_wait() {
+        // -1 === process was killed by a signal
+        Ok(Some(status)) => ProcessStatus::Finished(status.code().unwrap_or(-1)),
+        Ok(None) => ProcessStatus::Running,
+        Err(msg) => panic!("Error while trying to get process status: {}", msg),
```

It's better not to panic in Rust, but return Result. That's the beauty, you never panic and always know if function can return an error

---

Also, this will fail loudly bringing down all the other plugins. It's better to handle problems higher